gateway/middleware.go

```diff
@@ -692,17 +692,20 @@ func (t BaseMiddleware) CheckSessionAndIdentityForValidKey(originalKey *string,
 		return session.Clone(), true
 	}
 
-	t.Logger().Debug("Querying authstore")
-	// 2. If not there, get it from the AuthorizationHandler
-	session, found = t.Spec.AuthManager.KeyAuthorised(key)
-	if !found && storage.TokenOrg(key) != t.Spec.OrgID {
-		//treat it as a custom key
-		key = generateToken(t.Spec.OrgID, key)
-		cacheKey = key
-		if t.Spec.GlobalConfig.HashKeys {
-			cacheKey = storage.HashStr(cacheKey)
-		}
+	// Only search in RPC if not in emergency mode
+	if !rpc.IsEmergencyMode() {
```

Comment:
You make the assumption here that `AuthManager` is RPC layer. 
I would rather make it explicit, like: 
if ok := t.Spec.AuthManager.(RPCConnection); ok && rpc.IsEmergencyMode() {
   return
}

Reply:
@buger modified. Let me know what you think 👍